### PR TITLE
Python: Fix sample plugin path for Windows. Re-enable samples CI tests.

### LINKED ISSUE #6588: Python: Running Jupyter notebooks with pip package doesn't have proper cwd
**Describe the bug**
If running SK Python Jupyter notebooks after installing the pip package, the following import will fail:

`from samples.service_settings import ServiceSettings` due to a missing module error. Using the pip package alone without further directory configuration will cause this to fail.

**To Reproduce**
1. Go to a Jupyter notebook
2. Run the pip install (do not use Poetry as it will work)
3. Run the first few cells
4. See the module not found error

**Expected behavior**
The Jupyter notebooks should be able to run OOB.

**Screenshots**

**Platform**
 - OS: Any
 - IDE: VScode
 - Language: Python
 - Source: pip 1.0.5

### PR DESCRIPTION
**Pull request by moonbox3**
### Motivation and Context

Fix sample plugin path for Windows. Re-enable samples CI tests.

### Description

Fix sample plugin path for Windows. Re-enable samples CI tests.
- Tests passing locally on both MacOS and Windows.
- Adds logic to each notebook to properly set up the correct path to find imports. Notebooks should now run out of the box to close #6588
- Added retry logic for the samples tests to retry 3 times in the event of a failure.

The GH Action code quality check was failing for this same code in #6675, so this is superseding it.

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/f984fa2cd182b01baf3a0885983ab5c37bac0f14/README.md">Py3.10 Test Coverage</a><details><summary>Python 3.10 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td><b>TOTAL</b></td><td><b>6151</b></td><td><b>728</b></td><td><b>88%</b></td><td>&nbsp;</td></tr></tbody></table><i>report-only-changed-files is enabled. No files were changed during this commit :)</i></details>

## Python 3.10 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1434 | 1 :zzz: | 0 :x: | 0 :fire: | 25.246s :stopwatch: |

File: .github/workflows/python-integration-tests.yml [hunk 1]
```diff
@@ -103,6 +103,7 @@ jobs:
 
           cd python
           poetry run pytest ./tests/integration -v
+          poetry run pytest ./tests/samples -v
```
Comment (eavanvalkenburg) [inline .github/workflows/python-integration-tests.yml:106]: we also have a seperate workflow for samples, I'm wondering what will be better, this single one, or should (and can) we seperate them and have the samples be less of an issue when they don't go through?

File: python/samples/concepts/prompt_templates/configuring_prompts.py [hunk 2]
```diff
@@ -8,61 +8,76 @@
...
+kernel = Kernel()
 
-async def main():
-    kernel = Kernel()
+useAzureOpenAI = False
+model = "gpt-35-turbo" if useAzureOpenAI else "gpt-3.5-turbo-1106"
+service_id = model
 
-    useAzureOpenAI = False
-    model = "gpt-35-turbo" if useAzureOpenAI else "gpt-3.5-turbo-1106"
```
Comment (eavanvalkenburg) [inline python/samples/concepts/prompt_templates/configuring_prompts.py:16]: why point this at 1105, there is a newer one, or just point to latest (by using gpt-3.5-turbo): https://platform.openai.com/docs/models/gpt-3-5-turbo

File: python/samples/getting_started/00-getting-started.ipynb [hunk 3]
```diff
@@ -16,6 +16,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
+    "# Note: if using a Poetry virtual environment, do not run this cell\n",
     "!python -m pip install semantic-kernel==1.0.5"
```
Comment (eavanvalkenburg) [inline python/samples/getting_started/00-getting-started.ipynb:20]: we should probably move to this: https://ipython.readthedocs.io/en/stable/interactive/magics.html#magic-pip so %pip install which installs in the current kernel, rather then using `!` which executes in shell...

File: python/samples/getting_started/01-basic-loading-the-kernel.ipynb [hunk 4]
```diff
@@ -25,6 +25,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
+    "# Note: if using a Poetry virtual environment, do not run this cell\n",
     "!python -m pip install semantic-kernel==1.0.5"
```
Comment (eavanvalkenburg) [inline python/samples/getting_started/01-basic-loading-the-kernel.ipynb:29]: same as before

**Review by eavanvalkenburg — COMMENTED** [review]
couple of notes

File: .github/workflows/python-integration-tests.yml (hunk 1 above)
Comment (moonbox3) [reply .github/workflows/python-integration-tests.yml:106]: I think we need to place the same importance on samples (integration tests) as we do for normal integration tests. We should catch any errors right away and make sure they're fixed -- an error or breaking change, unless a service outage, should be handled ASAP.

File: python/samples/concepts/prompt_templates/configuring_prompts.py (hunk 2 above)
Comment (moonbox3) [reply python/samples/concepts/prompt_templates/configuring_prompts.py:16]: Will update, good catch

Comment (moonbox3) [reply python/samples/concepts/prompt_templates/configuring_prompts.py:16]: Updated all occurrences of `gpt-3.5-turbo-1106` to `gpt-3.5-turbo` which is `gpt-3.5-turbo-0125`.

File: python/samples/getting_started/00-getting-started.ipynb (hunk 3 above)
Comment (moonbox3) [reply python/samples/getting_started/00-getting-started.ipynb:20]: Great idea, let's do it. Will make the change now.

**Comment by moonbox3** [comment]
Closing as code is against a SK branch. Re-creating as #6691.

**Pull request closed without merging**